src/core/raster/qgsrasterinterface.cpp

```diff
@@ -188,10 +188,11 @@ QgsRasterBandStats QgsRasterInterface::bandStatistics( int bandNo,
         if ( blk->isNoData( i ) ) continue; // NULL
 
         double myValue = blk->value( i );
-
         myRasterBandStats.sum += myValue;
```

Comment:
@nyalldawson , should we also avoid adding infinity to the sum?

Reply:
Not sure... Probably? Maybe not... It'll make the stats meaningless either way, right - so maybe we are better to let users know this by returning a non finite number.

Reply:
@nyalldawson, right, that's what I thought too.